spec/models/zip_code_spec.rb

```diff
@@ -3,5 +3,75 @@
 require 'rails_helper'
 
 RSpec.describe ZipCode, type: :model do
-  it { should validate_presence_of(:d_codigo) }
+  context 'validations tests' do
+    it { should validate_presence_of(:d_codigo) }
+  end
+
+  context 'scopes tests' do
+    let(:params) do
+      { d_codigo: '01000',
+        d_asenta: 'San Ángel',
+        d_tipo_asenta: 'Colonia',
+        d_mnpio: 'Álvaro Obregón',
+        d_estado: 'Ciudad de México',
+        d_ciudad: 'Ciudad de México',
+        d_cp: '01001',
+        c_estado: '09',
+        c_oficina: '01001',
+        c_cp: '.',
+        c_tipo_asenta: '09',
+        c_mnpio: '010',
+        id_asenta_cpcons: '0001',
+        d_zona: 'Urbano',
+        c_cve_ciudad: '01' }
+    end
+
+    before(:each) do
+      ZipCode.create(params)
+    end
+
+    it 'should return a zip_code' do
+      expect(ZipCode.count).to eq(1)
+    end
+
+    it 'search for a wrong zip_code gives empty results' do
+      cp = '64000'
+      expect(ZipCode.find_by_zip_code(cp)).to eq([])
```

Comment:
Please follow the 4-phase testing way from thoughtbot - https://thoughtbot.com/blog/four-phase-test - in this case, maybe adding some spaces between the `exercise` & `verify` might help

Reply:
If I use `FactoryBot.create(:zip_code)` I don't need to save the model.
For example:
`create(:user)          # → model (saved)`
Taken from: [https://devhints.io/factory_bot](https://devhints.io/factory_bot)

Reply:
Yes, and sorry for not being clear enough, I meant, by adding some spaces between the `cp` variable and the `expect` directive, gives a better reading when you go through the code, something like:

```ruby
it 'search for a wrong zip_code gives empty results' do
  cp = '64000'
  
  expect(ZipCode.find_by_zip_code(cp)).to eq([])  
end
```

As you can see, by adding the space, it becomes more clear which step is the **exercise** and which one is the **verify**

Reply:
Done!